src/main/java/zmq/SocketBase.java

```diff
@@ -453,194 +453,221 @@ public final boolean connect(String addr)
         lock();
 
         try {
-            if (ctxTerminated) {
-                errno.set(ZError.ETERM);
-                return false;
-            }
+            return connectInternal(addr);
+        }
+        finally {
+            unlock();
+        }
+    }
 
-            options.mechanism.check(options);
+    public final int connectPeer(String addr)
```

Comment:
it seems like a big change, but I just changed connect to connectInternal and call it from connect and connectPeer.